consumer/src/main/kotlin/streams/kafka/KafkaEventSink.kt

```diff
@@ -113,15 +148,172 @@ class KafkaEventConsumer(private val consumer: KafkaConsumer<String, ByteArray>,
         StreamsUtils.ignoreExceptions({ consumer.close() }, UninitializedPropertyAccessException::class.java)
     }
 
-    override fun read(): Map<String, List<Any>>? {
-        val records = consumer.poll(config.sinkPollingInterval)
-        if (records != null && !records.isEmpty) {
-            return records
-                    .map {
-                        it.topic()!! to JSONUtils.readValue<Any>(it.value())
+    private fun readSimple(action: (String, List<Any>) -> Unit) {
+        val records = consumer.poll(0)
+        if (!records.isEmpty) {
+            try {
+                this.topics.forEach { topic ->
+                    action(topic, records.records(topic).map { JSONUtils.readValue<Any>(it.value()) })
+                }
+            } catch (e: Exception) {
+                // TODO add dead letter queue
+            }
+        }
+    }
+
+    private fun readFromPartition(config: KafkaTopicConfig, action: (String, List<Any>) -> Unit) {
+        setSeek(config.topicPartitionsMap)
+        val records = consumer.poll(0)
+        val consumerRecordsMap = toConsumerRecordsMap(config.topicPartitionsMap, records)
```

Comment:
why is the `records` handling here different from the readSimple method after `consumer.poll()`
one would guess it's the same kind of records.
I would kinda expect here only the setSeek as the only difference between the two methods?